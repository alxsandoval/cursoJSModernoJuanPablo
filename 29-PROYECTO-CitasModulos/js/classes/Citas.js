// Clases

// export default class Citas {
class Citas {

    constructor() {

        this.citas = [];

    }

    agregarCita ( cita ){

        this.citas = [...this.citas, cita];
        console.log( this.citas );
    }

    eliminarCita ( id ){

        // Obtenemos un nuevo arreglo que no tenda la cita con el id recibido
        this.citas = this.citas.filter( cita => cita.id !== id );

    }

    editarCita( citaActualizada ){

        // Map recorre los elementos del arreglo y nos retorna un nuevo arreglo
        this.citas = this.citas.map( cita => cita.id == citaActualizada.id ? citaActualizada : cita );
        
    }

};

// Existen 2 formas de exportar: Directamente en la clase o de esta manera
export default Citas;
