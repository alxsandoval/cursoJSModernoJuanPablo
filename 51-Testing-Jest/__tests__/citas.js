// Pruebas al proyecto de citas
// Para actualizar el snapshot: npm t -- -u

import Citas from "../js/classes/Citas";

// Grupo de pruebas
describe('Probar la clase de citas', ()=>{

    const citas = new Citas();
    const id = Date.now();

    // Prueba 1
    test('Agregar una nueva cita', ()=>{

        const citaObj = {
            id,
            mascota:     'Hook',
            propietario: 'Alexis',
            telefono:    '6644216047',
            fecha:       '21-12-2022',
            hora:        '12:05',
            sintomas:    'Solo duerme'
        };

        citas.agregarCita( citaObj );

        expect( citas ).toMatchSnapshot();

    });
    
    // Prueba 2
    test('Actualizar cita', ()=>{

        const citaActualizada = {
            id,
            mascota:     'NuevoNombre',
            propietario: 'Alexis',
            telefono:    '6644216047',
            fecha:       '21-12-2022',
            hora:        '12:05',
            sintomas:    'Solo duerme'
        };

        citas.editarCita( citaActualizada );

        expect( citas ).toMatchSnapshot();

    });

    // Prueba 3
    test('Eliminar cita', ()=>{

        citas.eliminarCita( id );

        expect( citas ).toMatchSnapshot();

    });


});
