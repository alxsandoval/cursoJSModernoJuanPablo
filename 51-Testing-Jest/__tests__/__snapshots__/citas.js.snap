// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "21-12-2022",
      "hora": "12:05",
      "id": 1671654004503,
      "mascota": "NuevoNombre",
      "propietario": "Alexis",
      "sintomas": "Solo duerme",
      "telefono": "6644216047",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "21-12-2022",
      "hora": "12:05",
      "id": 1671654004503,
      "mascota": "Hook",
      "propietario": "Alexis",
      "sintomas": "Solo duerme",
      "telefono": "6644216047",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
